### ISSUE TITLE
Backpropagation issue

### ISSUE THREAD

**Issue by ztnixon97** [tracel-ai/burn#2230]
**Describe the bug**
Opening at the request of Laggui. Back propagation on a Unet frozen on 0.14. 

**To Reproduce**
Model: https://github.com/ztnixon97/burn-regression-unet

**Expected behavior**
During Training the loss is calculated and the .backward() call is made. Gradients should be calculated instead the code hangs.

**Desktop (please complete the following information):**
 - OS: Windows 11 / WSL Ubuntu 22.03.3
 - Version 0.14.0

**Additional context**
in 0.13 learner returned this error:

== Tensor Operation Error ===
  Operation: 'Add'
  Reason:
    
The provided tensors have incompatible shapes. Incompatible size at dimension '1' => '64 != 512', which can't be broadcasted. Lhs tensor shape [1, 64, 64, 64], Rhs tensor shape [1, 512, 64, 64].

stack backtrace:
   0: std::panicking::beginpanic_handler
             at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library\std\src\panicking.rs:652
   1: core::panicking::panic_fmt
             at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library\core\src\panicking.rs:72
   2: core::array::drain::drain_array_with
   3: burn_tensor::tensor::api::numeric::<impl burn_tensor::tensor::api::base::Tensor<B,,K>>::add::paniccold_display
   4: burn_tensor::tensor::api::numeric::<impl burn_tensor::tensor::api::base::Tensor<B,,K>>::add
   5: burnautodiff::grads::Gradients::register
   6: <burn_autodiff::ops::base::OpsStep<B,T,SB,,_> as burn_autodiff::graph::base::Step>::step
   7: burn_autodiff::runtime::server::AutodiffServer::backward
   8: <burn_autodiff::runtime::mutex::MutexClient as burn_autodiff::runtime::client::AutodiffClient>::backward
   9: burn_unet::models::unet::UNetPlusPlus<B>::init
  10: burn_train::learner::epoch::TrainEpoch<TI>::run
  11: burn_core::record::recorder::Recorder::record
note: Some details are omitted, run with RUST_BACKTRACE=full for a verbose backtrace.
error: process didn't exit successfully: target\release\burn-unet.exe (exit code: 101)

in 0.14 and master no error is thrown and the training just hangs at the gradient calculation.

**Comment by laggui** [comment]
@ztnixon97 I see you closed the issue, did you figure out the problem?
